src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.PlatformServices/Options/IAppOptionsProvider.cs

```diff
@@ -0,0 +1,28 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Altinn.App.Common.Models;
+
+namespace Altinn.App.PlatformServices.Options
+{
+    /// <summary>
+    /// Interface for providing <see cref="AppOptions"/>
+    /// </summary>
+    public interface IAppOptionsProvider
+    {
+        /// <summary>
+        /// The id/name of the options this provider supports ie. land, fylker, kommuner.
+        /// You can have as many providers as you like, but you should have only one per
+        /// id. In the case of multiple providers registred with the same id it's non
+        /// deterministic which one will be returned.
```

Comment:
Seems like an obvious improvement to make duplicate registrations into an error.

Reply:
Yep, but I don't think I can prevent it from being added to the container. But I can add a check in the factory class and throw an exception in case it finds more than one with the same id. Would make it easier to avoid problems.